app/lib/crypto/hash.ts

```diff
@@ -18,3 +18,22 @@ const hmacFunction = (hashFunctionName: string, key: Buffer) => {
 export const sha512hmac = (key: Buffer, data: Buffer) => {
   return hmacFunction("sha512", key)(data)
 }
+
+/**
+ * Create sha function for the indicated hash function
+ * @param {string} hashFunctionName name of a hash function
+ * @param {Buffer} bytes
+ * @returns {Buffer} sha function
+ */
+const shaFunction = (hashFunctionName: string, bytes: Buffer) => {
```

Comment:
This function name isn't as expressive as desirable, we should always favor actions/verbs over nouns.

What about `applySHA()`?

Reply:
Or even just `applyHash()` or `hash()`, as it accepts other algorithms out of the SHA family.